woocommerce/woocommerce-ios
# Identify language using a separate AI prompt

Part of: #10103


## Description

Tries to identify the language using a separate prompt. Inputs the identified language to the text generation prompt.

### Notes
I think we should skip storing the identified language due to the following reasons.
1. A store might have products in different languages.
2. If incorrect language is identified for a product the first time, we will keep generating text in the wrong language for all other products. (which will be a pretty bad user experience)

Should we cache the language information to reduce network requests?

## Testing instructions

- Log in to a published WPCom site.
- Select an existing product or create a new one.
- Select Write with AI in the description row.
- On the description AI sheet, select Write with AI. The response should be in the same language as the product title and features.
- Tap Regenerate and make sure that new responses are still good.
- Tap the `...` button from the top right and tap Share to open the share sheet
- Tap "Write with AI" to generate a "share message".
- The generated share message should be in the same language as the product title and description.


## Screenshots
| Product description | Product sharing |
|--------|--------|
| ![Simulator Screen Shot - iPhone 14 Pro - 2023-07-05 at 10 28 26](https://github.com/woocommerce/woocommerce-ios/assets/524475/92c8129a-a6e0-445d-9cf1-0478778b8169) | ![Simulator Screen Shot - iPhone 14 Pro - 2023-07-05 at 10 28 15](https://github.com/woocommerce/woocommerce-ios/assets/524475/fd538127-befe-42cf-9c88-62fb9fd16da6) |



---
